### PR TITLE
Feature/alerting lambda to timestream

### PR DESCRIPTION
**Pull request by natmasslove**

### CODE REVIEW

File: infra_tooling_account/infra_tooling_account/infra_tooling_common_stack.py [hunk 1]
```diff
@@ -96,6 +116,21 @@ def create_timestream_db(self):
 
         return timestream_storage
 
+    def create_timestream_tables(self, timestream_storage):
+        retention_properties_property = timestream.CfnTable.RetentionPropertiesProperty(
+            magnetic_store_retention_period_in_days="365",
```
Comment (nalkhimo) [inline infra_tooling_account/infra_tooling_account/infra_tooling_common_stack.py:121]: Wouldn't be better to move these contants "365" and "240" into settings or contants.py file?

File: src/lib/aws/timestream_manager.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        # Convert the ISO date string to a datetime object
+        dt = datetime.fromisoformat(iso_date.rstrip("Z"))
+        # Convert the datetime object to epoch time in seconds
+        epoch_time = dt.timestamp()
+
+    # Convert epoch time to milliseconds and return as string
+    return str(int(epoch_time * 1000))
+
+
+class TimestreamTableWriter:
```
Comment (nalkhimo) [inline src/lib/aws/timestream_manager.py:33]: What do you think about renaming this class to TimestreamManager? And in future we will be able to add new methods connected to Timestream if needed.

File: src/lib/aws/timestream_manager.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        Args:
+            records: A list of records to be written to the Timestream table.
+
+        Returns:
+            The response from the Timestream write_records API call.
+
+        Todo:
+            Introduce records buffering to support batches of less than 100 records.
+        """                
+        result = self.timestream_write_client.write_records(
```
Comment (nalkhimo) [inline src/lib/aws/timestream_manager.py:80]: Better to add try-except block with a custom exception. That will help in future to understand where error occurred.

**Review by nalkhimo — APPROVED** [review]

**Review by NastassiaPaskrobka — APPROVED** [review]

File: infra_tooling_account/infra_tooling_account/infra_tooling_common_stack.py (hunk 1 above)
Comment (natmasslove) [reply infra_tooling_account/infra_tooling_account/infra_tooling_common_stack.py:121]: Agree. Added trello task for future not to bloat current PR

File: src/lib/aws/timestream_manager.py (hunk 2 above)
Comment (natmasslove) [reply src/lib/aws/timestream_manager.py:33]: Actually, that was the idea to name like that.
In future we will have, I guess, TimestreamTableReader, which is significantly different.
It even uses another boto3 client (there are separate clients for timestream write and read).

File: src/lib/aws/timestream_manager.py (hunk 3 above)
Comment (natmasslove) [reply src/lib/aws/timestream_manager.py]: fixed

**Review by vdektiarev — APPROVED** [review]

File: infra_tooling_account/infra_tooling_account/infra_tooling_alerting_stack.py [hunk 4]
```diff
@@ -150,7 +166,7 @@ def create_alerting_lambda(
 
         alerting_lambda_role.add_to_policy(
             iam.PolicyStatement(
-                actions=["queue:SendMessage"],
```
Comment (vdektiarev) [inline infra_tooling_account/infra_tooling_account/infra_tooling_alerting_stack.py:153]: My bad, looked in the wrong place for permission names references. Good catch!

**Pull request merged**